app/components/rdf-form-fields/rich-text-editor.js

```diff
@@ -116,16 +116,16 @@ export default class RdfFormFieldsRichTextEditorComponent extends SimpleInputFie
       // Only trigger an update if the value actually changed.
       // This prevents that the form observer is triggered even though no editor content was changed.
       if (this.value !== editorValue) {
-        this.updateValueInStore([editorValue]);
+        this.updateValueInStore(editorValue);
       }
     }
   }
 
   @action
-  updateValueInStore(values) {
-    this.value = values[0] || '';
-    super.updateValue(this.value);
-    this.setInitialValue();
```

Comment:
This was the root of the problem. The `updateValueInStore` method was also called by the `updateValue` method, which in turn was called on "focusout" of the editor window. So everytime the editor loses focus, we were calling 'setInitialValue` which also moves the cursor.

We now only call `setInitialValue` (renamed to `setEditorValue`) when the component is first rendered, or when the user selects one of the three-way comparison options.